### PR TITLE
sunraster JOSS paper.

### PR DESCRIPTION
**Pull request by DanRyanIrish**
## PR Description

This PR presents a sunraster paper for submission to the Journal of Open Source Software.

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
@nabobalis @Cadair @BaptistePellorceAstro @ehsteve @abit2 @tiagopereira @ebuchlin
Thank you for agreeing to be co-authors of this JOSS paper on sunraster.  Here is the latest draft.  As is normal with JOSS submissions it is a short summary (~a page) of the package and the need it fulfils.  The review will focus largely on the code-base.  Please feel free to provide any feedback.  Otherwise I hope to submit on Friday.

File: joss_paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+sunraster is a free, open-source, community-developed Python package for inspecting,
+manipulating, and visualizing solar rastering slit-spectrograph observations.
+The sunraster data classes link observations with various forms of supporting data
+including: measurement uncertainties; units; a data mask to mark pixels with unreliable
+or un-physical data values; WCS (World Coordinate System) transformations that describe
+the position, wavelengths and times represented by the pixels; and general metadata.
+These classes provide support for analysis tasks, such as extracting regions of
+interest and applying/removing exposure time corrections, as well as a quick-look
+visualization suite.
+sunraster depends heavily on the ndcube package [@ndcube-apj, @ndcube-joss] and its
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:72]:
```suggestion
sunraster depends heavily on the ndcube package [@ndcube-apj; @ndcube-joss] and its
```

**Review by ebuchlin — COMMENTED** [review]
Some comments on the paper...

File: joss_paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+for seamlessly switching between the 3-D and 4-D paradigms without changing or
+hard-copying the underlying data.
+This saves users carrying around multiple copies of the same data which might become
+inconsistent during a data analysis workflow.
+A powerful slicing API enables users to extract regions of interest with single commands
+that not only alter the data array, but also the associated uncertainties, data mask
+and the coordinate transformations.
+This saves users from performing multiple tedious and error-prone tasks on the various
+compoennts of their dataset individually.
+This makes the development of analysis workflows quicker, more reliable, and easier
```
Comment (takucera) [inline joss_paper/paper.md:112]: "It also makes"
```suggestion
This makes the development of analysis workflows quicker, more reliable, and easier
```

File: joss_paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+for their analysis.
+In addition, the Python user tools package for the IRIS mission, irispy-lmsal
+[irispy-docs; @irispy-code], depends on sunraster.
+
+# Acknowledgements
+
+We acknowledge financial support for sunraster from Solar Orbiter/SPICE
+(grant 80NSSC19K1000).
+We also acknowledge the SunPy and Python in Heliophysics communities for their support.
+
```
Comment (nabobalis) [inline joss_paper/paper.md:129]:
```suggestion
We acknowledge financial support for sunraster from NASA's SDO/AIA and IRIS missions
(grant NNG04EA00C and NNG09FA40C).
```

File: joss_paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+These classes provide support for analysis tasks, such as extracting regions of
+interest and applying/removing exposure time corrections, as well as a quick-look
+visualization suite.
+sunraster depends heavily on the ndcube package [@ndcube-apj, @ndcube-joss] and its
+SpectrogramCube class complies with the NDCube 2 API [sep12; @ndcube-apj].
+
+# Statement of Need
+
+Observing the location, morphology, dynamics and energetics of physical phenomena
+are often key to understanding the underlying physics that drives them.
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply joss_paper/paper.md:78]: Indeed you are right!

```suggestion
is often key to understanding the underlying physics that drives them.
```

File: joss_paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+However, making simultaneous spatial, temporal and spectral measurements with a single
+instrument is often challenging.
+One solution is to disperse the incoming light through a narrow slit onto a pixelated
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:81]:
```suggestion
One option is to disperse the incoming light through a narrow slit onto a pixelated
```

File: joss_paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+that not only alter the data array, but also the associated uncertainties, data mask
+and the coordinate transformations.
+This saves users from performing multiple tedious and error-prone tasks on the various
+compoennts of their dataset individually.
+This makes the development of analysis workflows quicker, more reliable, and easier
+to understand, thus boosting users' productivity.
+
+# Community Applications
+
+Most of sunrasters tools are instrument-agnostic.
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply joss_paper/paper.md:117]:
```suggestion
Most of sunraster's tools are instrument-agnostic.
```

File: joss_paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+However it does provide specific tools for reading Solar Orbiter/SPICE FITS files
+and hence helping SPICE users leverage the wider scientific Python ecosystem
+for their analysis.
+In addition, the Python user tools package for the IRIS mission, irispy-lmsal
+[irispy-docs; @irispy-code], depends on sunraster.
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:122]: How about this @ebuchlin 
```suggestion
In addition, the data and metadata classes in the IRIS mission's Python user tools package [irispy-lmsal;
@irispy-docs; @irispy-code] inherit from and build upon those provided by sunraster.
```

File: joss_paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+# Acknowledgements
+
+We acknowledge financial support for sunraster from Solar Orbiter/SPICE
+(grant 80NSSC19K1000).
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:127]:
```suggestion
(grant 80NSSC19K1000) as well as NASA's SDO/AIA and IRIS missions
(grant NNG04EA00C and NNG09FA40C).
```

File: joss_paper/paper.md (hunk 3 above)
Comment (DanRyanIrish) [reply joss_paper/paper.md:129]: Moving this to be part of first sentence.

File: joss_paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+By building on features inherited from the ndcube package, sunraster provides an API
+for seamlessly switching between the 3-D and 4-D paradigms without changing or
+hard-copying the underlying data.
+This saves users carrying around multiple copies of the same data which might become
+inconsistent during a data analysis workflow.
+A powerful slicing API enables users to extract regions of interest with single commands
+that not only alter the data array, but also the associated uncertainties, data mask
+and the coordinate transformations.
+This saves users from performing multiple tedious and error-prone tasks on the various
+compoennts of their dataset individually.
```
Comment (nabobalis) [inline joss_paper/paper.md:111]:
```suggestion
components of their dataset individually.
```

**Review by nabobalis — COMMENTED** [review]
LGTM

**Comment by DanRyanIrish** [comment]
pre-commit.ci autofix

**Comment by DanRyanIrish** [comment]
@nabobalis @Cadair @BaptistePellorceAstro @ehsteve @abit2 @tiagopereira @ebuchlin @takucera : Thank you to those who have provided comments.  I believe we are ready to submit.  I will do so tonight or tomorrow morning unless people need more time to provide feedback.

**Comment by DanRyanIrish** [comment]
I have just submitted this paper to JOSS.  Thanks for all your input.

**Pull request merged**